sdk/src/client/api/block_builder/input_selection/mod.rs

```
@@ -420,6 +430,10 @@ impl InputSelection {
 
         self.validate_transitions()?;
 
+        // Only keep mana rewards for selected inputs
```

But why do we have more to start with?

---

We shouldn't, but the user could pass in whatever so I figured I'd check

---

Wouldn't it be better to return an error in this case?

---

I suppose

---

done